composables/useMultipleBalance.ts

```diff
@@ -35,139 +33,141 @@ const getNetwork = (prefix: Prefix) => {
+    : await balanceOf(api, prefixAddress)
+  const currentBalance = format(nativeBalance, chain.tokenDecimals, false)
+  await wsProvider.disconnect()
+
+  return { nativeBalance, currentBalance }
+}
 
-    const api = await ApiPromise.create({
-      provider: wsProvider,
-    })
+const getBalance = async (chainName: string, token = 'KSM', tokenId = 0) => {
+  const { accountId } = useAuth()
 
-    const nativeBalance = await getNativeBalance({
-      address: prefixAddress,
-      api: api,
-      tokenId,
-    })
+  const currentAddress = accountId.value
+  const prefix = networkToPrefix[chainName]
+  const chain = CHAINS[prefix]
 
-    const currentBalance = format(nativeBalance, chain.tokenDecimals, false)
+  const defaultAddress = toDefaultAddress(currentAddress)
+  const publicKey = decodeAddress(currentAddress)
+  const prefixAddress = encodeAddress(publicKey, chain.ss58Format)
 
-    let selectedTokenId = String(tokenId)
-    if (chainName === 'basilisk') {
-      selectedTokenId = await getAssetIdByAccount(api, prefixAddress)
-    }
+  const balances = await getCombinedBalances(chainName, tokenId, prefixAddress)
 
-    const usd = calculateUsd(
-      currentBalance,
-      fiatStore.getCurrentTokenValue(token),
-    )
-
-    identityStore.setMultiBalances({
-      address: defaultAddress,
-      chains: {
-        [chainName]: {
-          [token.toLowerCase()]: {
-            address: prefixAddress,
-            balance: currentBalance,
-            nativeBalance,
-            usd,
-            selected: selectedTokenId === String(tokenId),
-          },
+  return {
+    address: defaultAddress,
+    chains: {
+      [chainName]: {
+        [token.toLowerCase()]: {
+          address: prefixAddress,
+          balance: balances.currentBalance,
+          nativeBalance: balances.currentBalance,
+          usd: calculateUsd(balances.currentBalance, token),
+          selected: chainName === 'basilisk',
         },
       },
-      chainName,
-    })
-
-    identityStore.multiBalanceNetwork = currentNetwork.value
-
-    return wsProvider.disconnect()
+    },
+    chainName,
   }
+}
 
-  const fetchFiatPrice = async (force) => {
-    if (!force && fiatStore.incompleteFiatValues) {
-      await fiatStore.fetchFiatPrice()
-    }
+const fetchFiatPrice = async (force) => {
+  const fiatStore = useFiatStore()
+  if (!force && fiatStore.incompleteFiatValues) {
+    await fiatStore.fetchFiatPrice()
   }
+}
 
-  const fetchMultipleBalance = async (
-    onlyPrefixes: Prefix[] = [],
-    forceFiat: boolean = false,
-  ) => {
-    await fetchFiatPrice(forceFiat)
-
-    const assets = isTestnet.value
-      ? multiBalanceAssetsTestnet.value
-      : multiBalanceAssets.value
-
-    const chainNetworks = onlyPrefixes.map(getNetwork).filter(Boolean)
-
-    const assetsToFetch = onlyPrefixes.length
-      ? assets.filter((item) => chainNetworks.includes(item.chain))
-      : assets
+const fetchMultipleBalance = async (
+  currentNetwork,
```

Comment:
1. Why should we add such a params? We could get it directly from `usePrefix`.

2. As you change the input parameters of this function, it would break somewhere this function is called.

<img width="389" alt="image" src="https://github.com/kodadot/nft-gallery/assets/31397967/f1dc0132-26bf-4c37-8f7b-def06b214b31">